python/beeai_framework/adapters/watsonx_orchestrate/serve/_factories/_requirement_agent.py

```diff
@@ -24,11 +24,20 @@ class WatsonxOrchestrateServerRequirementAgent(WatsonxOrchestrateServerAgent[Req
     def model_id(self) -> str:
         return self._agent._llm.model_id
 
-    async def _run(self) -> AssistantMessage:
-        response = await self._agent.run(prompt=None)
-        return response.answer
+    async def _run(self, input: list[AnyMessage]) -> AssistantMessage:
+        if input or not self._agent.memory.is_empty():
+            input = self._agent.memory.messages[-1:]
+            response = await self._agent.run(input)
+            return response.message
+        else:
+            raise ValueError("Agent invoked with empty memory.")
```

Comment:
Why did you change the signature of the function? It should be kept as is. If you read the flow carefully, you will see the memory is set before the `run` is called. Please revert your changes, as this completely breaks the agent.

Reply:
https://github.com/i-am-bee/beeai-framework/blob/df8a4c0b49d6a48aca568097411390b017faaf83/python/beeai_framework/adapters/watsonx_orchestrate/serve/api.py#L78-L90